test/functional/homepage_controller_test.rb

```diff
@@ -76,8 +78,23 @@ class HomepageControllerTest < ActionController::TestCase
       assert_equal new_url, PopularLinksEdition.last.link_items[0][:url]
     end
 
+    should "update publishing API" do
+      Services.publishing_api.expects(:put_content)
+
+      patch :update, params: { id: @popular_links.id,
+                               "popular_links" =>
+                                 { "1" => { "title" => "title", "url" => "url.com" },
+                                   "2" => { "title" => "title2", "url" => "https://www.url2.com" },
+                                   "3" => { "title" => "title3", "url" => "https://www.url3.com" },
+                                   "4" => { "title" => "title4", "url" => "https://www.url4.com" },
+                                   "5" => { "title" => "title5", "url" => "https://www.url5.com" },
+                                   "6" => { "title" => "title6", "url" => "https://www.url6.com" } } }
+    end
+
     should "redirect to show path on success" do
+      UpdateService.stubs(:call)
```

Comment:
Do we need this stubbing? Commenting it out doesn't seem to cause any ill-effect.